packages/liveblocks-react/src/room.tsx

```diff
@@ -1216,12 +1216,40 @@ export function createRoomContext<
       [room, queryKey] // eslint-disable-line react-hooks/exhaustive-deps
     );
 
-    return useSyncExternalStoreWithSelector(
+    const state = useSyncExternalStoreWithSelector(
       store.subscribe,
       store.get,
       store.get,
       selector
     );
+
+    React.useEffect(
+      () => {
+        if (state.isLoading) return;
+
+        const hash = window.location.hash;
+        const commentId = hash.slice(1);
+
+        if (!commentId.startsWith("cm_")) return;
+
+        const comments = state.threads.flatMap((thread) => thread.comments);
+        const isCommentInThreads = comments.some(
+          (comment) => comment.id === commentId
+        );
+
+        // If the comment is not in the threads for this hook, we do not scroll to it
+        if (!isCommentInThreads) return;
+
+        const comment = document.getElementById(commentId);
+        if (comment === null) return;
```

Comment:
Should we reverse this logic? Start by `getElementById` **then** make the `cm_`/state checks? I feel like `getElementById` is cheaper than the state check, so if you use primitives (they won't have comment `id`s) or for some reasons have replaced and changed comment `id`s, we would not run the state checks.

Or maybe sandwich it: start with the cheapest `cm_` check, then `getElementById`, and finally the state check.